### PR TITLE
NEXUS-23681 Hosted metadata generation for Helm

### PR DESCRIPTION
**Pull request by rharaiev**
Ticket: https://issues.sonatype.org/browse/NEXUS-23681

### CODE REVIEW

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/HelmContentFacet.java [hunk 1]
```diff
@@ -30,6 +33,8 @@
 public interface HelmContentFacet
     extends ContentFacet
 {
+  Iterable<FluentAsset> browseComponentAssets(@Nullable final AssetKind assetKind);
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/HelmContentFacet.java:36]: I don't see any components involved here...

Comment (chernikovmaksym) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/HelmContentFacet.java:36]: Maybe something like `browseAssetByAssetKind()`?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import static org.sonatype.repository.helm.internal.database.HelmProperties.VERSION;
+
+/**
+ * Build index.yaml file for Helm Hosted
+ *
+ * @since 1.0.15
+ */
+@Named
+@Singleton
+public class CreateIndexServiceImpl
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java:56]: Could you create a ticket for adding event handlers and put its number in the todo here?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      chartEntry.setDigest(blob != null ? blob.checksums().get("sha256") : null);
+      createListOfRelativeUrls(formatAttributes, chartEntry);
+      chartEntry.setSources(formatAttributes.get(SOURCES.getPropertyName(), List.class));
+      index.addEntry(chartEntry);
+    }
+  }
+
+  private void createListOfRelativeUrls(final NestedAttributesMap formatAttributes, final ChartEntry chartEntry) {
+    List<String> urls = new ArrayList<>();
+    urls.add(String.format("%s-%s.tgz",
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java:111]: Minor: we used to create constants for string formatting in other formats

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+  private final YamlParser yamlParser;
+
+  @Inject
+  public CreateIndexServiceImpl(final YamlParser yamlParser) {
+    this.yamlParser = checkNotNull(yamlParser);
+  }
+
+  @Nullable
+  public Content buildIndexYaml(final Repository repository) {
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java:72]: Do we have test coverage for the metadata?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/HelmContentFacet.java (hunk 1 above)
Comment (chernikovmaksym) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/HelmContentFacet.java:36]: oh, I see you just copied it from the orient version. Okay, naming is up to you 😄

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java (hunk 4 above)
Comment (rharaiev) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java:72]: nope

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java (hunk 3 above)
Comment (rharaiev) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java:111]: just copied method from `CreateIndexServiceImpl` class for Orient

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java (hunk 4 above)
Comment (chernikovmaksym) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java:72]: now we have it. Kinda have it😆

**Pull request merged**